src/main/java/uk/gov/hmcts/ccd/domain/service/DocLinksRestoreService.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.hmcts.ccd.domain.service;
+
+import org.apache.commons.collections.ListUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.casedetails.CaseAuditEventEntity;
+import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+
+@Component
+public class DocLinksRestoreService {
+    private static final Logger LOG = LoggerFactory.getLogger(DocLinksRestoreService.class);
+
+    public static final String START_TIME = "'2019-08-20 15:50:00.000'";
+    public static final String END_TIME = "'2019-08-21 13:30:00.000'";
+    public static final String BETWEEN_CLAUSE = "BETWEEN  " + START_TIME + " AND " + END_TIME;
+
+    public static final String OLD_CASE_QUERY = "SELECT * FROM case_data WHERE last_modified " + BETWEEN_CLAUSE + " AND created_date < " + START_TIME;
```

Comment:
OLD_CASE_QUERY : aren't we skipping cases that have events in the bug period but that have then been modified after the bug period? Those should be included as well.

Reply:
I think the correct way would be to find the id of any case which has any event in the bug timeframe.

Reply:
Yes, that's right.